src/base/http/connection.cpp

```diff
@@ -91,7 +94,7 @@ void Connection::read()
             return;
 
         case RequestParser::ParseStatus::OK: {
-                const Environment env {m_socket->localAddress(), m_socket->localPort(), m_socket->peerAddress(), m_socket->peerPort()};
+                const Environment env {m_socket->localAddress(), m_socket->localPort(), ReverseProxy::resolvPeerAddress(m_socket, result.request), m_socket->peerPort()};
```

Comment:
you shouldn't change this definition, IMO this is for the network layer, don't taint app layer with it.

Reply:
It was my original belief and I am becoming more and more established in it. It shouldn't be substituted.

Reply:
>If you think the patch will never be included in official repo, say so, we both will avoid wasting more time.

For a PR to get merged you need at least 2 approval from team members.

This is a bit embarrassing, everyone you see here is a mere contributor (except @sledgehammer999, the project owner) to this project, we are limited to only express our personal viewpoints... AFAIK we never got the rights to turn down PRs and sometimes I wish we do, to save everyone's time.

>I just thought it may interest someone else and I want to bring my modest help.

It is very kind of you. Just sharing my viewpoint, doing changes to the official/upstream repo means your code (most of the time) will be maintained by the maintainers. If the changes benefits everyone, it is more than welcome and others will be interested in help maintaining it, the feature will live on forever if not superseded; however if it is the opposite, then it might decay over time (e.g. our DDNS feature) and it might got removed in a future review (by future maintainers).

Reply:
IMHO, having opinion of other contributors is important because as contributor, you are able to share your past experience and point out problems you have met or notice problems linked to your own usage.

> If the changes benefits everyone, it is more than welcome and others will be interested in help maintaining it, the feature will live on forever if not superseded; however if it is the opposite, then it might decay over time (e.g. our DDNS feature) and it might get removed in a future review (by future maintainers).

It is always the risk. Having the code added to the offical repo avoids at least creation of an external patch. Such a patch can be hard to update to take into account official code changes.